xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -243,7 +245,9 @@ void CGUIWindowManager::CreateWindows()
 
   Add(new CGUIDialogInfoProviderSettings);
 
+#ifdef HAS_DVD_DRIVE
   Add(new CGUIDialogPlayEject);
+#endif
```

Comment:
You should also ifdef the respective code to destroy the window, located also in this source file => `DestroyWindow(WINDOW_DIALOG_PLAY_EJECT);`